core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -168,11 +177,12 @@
 
     protected final Set<Startable> dependencies = new HashSet<>();
 
-    /*
+    /**
      * Unique instance of DockerClient for use by this container object.
+     * We use {@link LazyDockerClient} here to avoid eager client creation
      */
     @Setter(AccessLevel.NONE)
-    protected DockerClient dockerClient = DockerClientFactory.instance().client();
+    protected DockerClient dockerClient = LazyDockerClient.INSTANCE;
```

Comment:
needed this one for unit tests, plus good improvement in general. Was also reported as https://github.com/testcontainers/testcontainers-java/issues/1749